src/logic/safe/utils/maxPriorityFeePerGas.ts

```diff
@@ -0,0 +1,34 @@
+import { hexToNumber } from 'web3-utils'
+
+import { getWeb3ReadOnly } from 'src/logic/wallets/getWeb3'
+
+export const clampMaxPrioFeePerGas = (maxPriorityFeePerGas: number, maxFeePerGas: number): number => {
+  if (maxPriorityFeePerGas > maxFeePerGas) {
+    const MIN_MAX_PRIO_FEE = 0.5
+    const MAX_MAX_PRIO_FEE = maxFeePerGas - 1
+
+    return Math.max(MIN_MAX_PRIO_FEE, MAX_MAX_PRIO_FEE)
+  }
+
+  return maxPriorityFeePerGas
+}
+
+const DEFAULT_MAX_PRIO_FEE = 2.5e9 // 2.5 GWEI
+export const getMedianMaxPrioFee = async (): Promise<number> => {
```

Comment:
I would suggest this function to only return `maxPriorityFeePerGas` and do the clamp outside to reduce complexity

Reply:
So, would you suggest returning`maxFeePerGas` and `maxPriorityFeePerGas` here then? I would need them to calculate the clamping

Reply:
We could return the raw `maxPriorityFeePerGas` from this function and clamp when needed e.g. when getting the gas price estimate from an oracle.